### ISSUE TITLE
My Reps Screen Blank

### ISSUE THREAD

**Issue by jterps08** [PowerlineApp/powerline-mobile#418]
New build erik+develop... My Reps Screen is blank

**Comment by petervojtek** [comment]
@jterps08 , I am not sure what do you expect to see -- some representatives, or a message saying "you have no representatives" ?

For Peter10 the list of his representatives is an empty array:

```
Request URL:https://api-dev.powerli.ne/api/representatives/
Request Method:GET
Status Code:200 OK

[]
```

**Comment by jterps08** [comment]
@petervojtek this screen is supposed to show representatives. I have provided a real address for my username 1.JesseChen and nothing is showing up on Representatives screen. This was previously working before Erik's changes were included I believe or at least a few builds ago it was working...

If there are no Representatives, it can say "We can link you to your Elected Representatives in the US, CA, MX, GB, AU, and NZ only. We also support linking to local elected leaders worldwide." 

But, for the bug above, there should be results showing and there are none for me.

**Comment by petervojtek** [comment]
@jterps08 , the issue is I cannot login as `1.JesseChen` (I guess it is your facebook user) and thus I am unable to reproduce this issue

**Comment by petervojtek** [comment]
@igaponov , currenly the request to get representatives seem to fail entirely:

```
Request URL:https://api-dev.powerli.ne/api/representatives/
Request Method:GET
Status Code:500 Internal Server Error

{"code":500,"message":"An exception occurred while executing 'SELECT r0_.id AS id0, r0_.avatar_file_name AS avatar_file_name1, r0_.avatar_source_file_name AS avatar_source_file_name2, r0_.officialTitle AS officialTitle3, r0_.country AS country4, r0_.city AS city5, r0_.address1 AS address16, r0_.address2 AS address27, r0_.address3 AS address38, r0_.phone AS phone9, r0_.privatePhone AS privatePhone10, r0_.status AS status11, r0_.email AS email12, r0_.privateEmail AS privateEmail13, r0_.fax AS fax14, r0_.website AS website15, r0_.party AS party16, r0_.birthday AS birthday17, r0_.start_term AS start_term18, r0_.end_term AS end_term19, r0_.facebook AS facebook20, r0_.youtube AS youtube21, r0_.twitter AS twitter22, r0_.openstate_id AS openstate_id23, r0_.cicero_id AS cicero_id24, r0_.questions_limit AS questions_limit25, r0_.is_nonlegislative AS is_nonlegislative26, r0_.updatedAt AS updatedAt27, r0_.user_id AS user_id28, r0_.local_group AS local_group29, r0_.state AS state30, r0_.district_id AS district_id31 FROM representatives r0_ WHERE r0_.is_nonlegislative = 1 AND r0_.district_id IN ()':\n\nSQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')' at line 1","errors":null}
```

**Comment by igaponov** [comment]
@petervojtek I'm rebuilding representatives now and didn't know that app uses this endpoint. The problem is fixed, but representative's data was changed so you need to check if app is working with new structure.

**Comment by jterps08** [comment]
@igaponov - Please download a production version of the app from the app stores and reference how the app constructs this screen currently. I want you to see how it is supposed to work. I know you are working on the Rep concept for creating leader content and this accident was going to occur eventually... but we need to make sure that My Reps screen is working the way it was just last week. 

@petervojtek - Please simply register a new user and use the address 25 Broadway New York New York 10004... That should give you full representatives in this screen. Local state and national.

**Comment by petervojtek** [comment]
@jterps08 , I registered new user (`Peter50`) with address `25 Broadway New York New York 10004`, but I receive empty list of representatives:

```
Request URL:https://api-dev.powerli.ne/api/representatives/
Request Method:GET
Status Code:200 OK

[]
```

<img width="426" alt="screenshot 2016-11-23 11 17 54" src="https://cloud.githubusercontent.com/assets/225506/20558053/92302ea2-b16e-11e6-9007-680dc96173cf.png">

@igaponov , according to Jesse's comment above, I should see receive some representatives for this user.

**Comment by igaponov** [comment]
@petervojtek this hint works only for production. There are no representatives on dev.

**Comment by jterps08** [comment]
@igaponov - This should work in production AND in dev. It has worked in dev the entire last few months. Again, please roll back your changes and restore this functionality to how it worked before. 

1. My Representatives shows a list of all of the user's elected leaders regardless of whether they are registered on Powerline or not.
2. My Representatives screen DOES NOT show the user only the elected leaders who have registered on Powerline. That is not the intent of this screen. 
3. My Representatives screen DOES show the user all the elected leaders that can be found for him based on Cicero lookup. 

I want this to work *exactly* how it was working before you started refactoring Representatives please. If Jesse lives in New York and Igor lives in New York, and Jesse registers on Powerline first, then backend will lookup Jesse's Representatives using Cicero API credits, and store on backend. All 'extra' information about the representatives (office number, website, etc etc) is provided via Cicero and stored as well on backend... When Igor registers on Powerline (after Jesse), system looks up Igor's representatives (I believe using Cicero), but does not need to use any extra API credits to look up extra information. We store as much information on backend so that we can reduce the amount of API credits we need to leverage when we are getting this information.

Contact me on Slack if you have any questions...

In short --> This endpoint needs to work the way it used to work.

**Comment by igaponov** [comment]
@jterps08 ok, now it's clear. I thought we have only representatives that were linked to registered users.

**Comment by jterps08** [comment]
Yup. Thanks Igor. I know this is hard to explain via written text so I appreciate your patience.

**Comment by jterps08** [comment]
@igapanov any chance you can restore this endpoint to previous working behavior this morning? I have a meeting in a couple hours where I'd like this to work if possible

**Comment by igaponov** [comment]
@jterps08 I can't just revert it now, there were many changes after this one. I can deploy previous build, but not sure it will work with the last version of app.

**Comment by jterps08** [comment]
OK. We need that functionality back to working as soon as possible. Keep
your current code but please make this a top priority. We can't go live
without it and I didn't expect it to be impacted by the elected leader work.

**Comment by igaponov** [comment]
Reverted.
I've added districts for user `Peter50`, new users will get districts after registration. I can add districts for old users if it's needed.

**Comment by petervojtek** [comment]
@igaponov , the list of representatives is provided successfully for Peter50, but when I a about to view details of some particual representative, I face this (the request is from legacy code, I have not changed it.. but there does not seem any v2 endpoint I can switch to):

```
Request URL:https://api-dev.powerli.ne/api/representatives/info/68851/0
Request Method:GET
Status Code:404 Not Found
```

For Peter10 I face following problem:

```
Request URL:https://api-dev.powerli.ne/api/representatives/
Request Method:GET
Status Code:500 Internal Server Error

{"code":500,"message":"[Syntax Error] line 0, col 81: Error: Expected Literal, got ')'","errors":null}
```

**Comment by igaponov** [comment]
@petervojtek the error is fixed.
@jterps08 should we support the old api or add v2 endpoints for representatives?

**Comment by jterps08** [comment]
we can support old API @igaponov ... no need to build new endpoints to support same functionality.

**Comment by petervojtek** [comment]
@igaponov , currrently viewing representative detail fails with error 404 (i tried several representatives):

```
Request URL:https://api-dev.powerli.ne/api/representatives/info/103108/0
Request Method:GET
Status Code:404 Not Found
```

**Comment by igaponov** [comment]
@petervojtek the right link is `https://api-dev.powerli.ne/api/representatives/info/0/103108`

The first id means representative registered in our system (`id` attribute in the result of `https://api-dev.powerli.ne/api/representatives/`), the second id is representative from cicero db (`storage_id` attribute)

**Comment by petervojtek** [comment]
@igaponov ,  the url you provided does not seem to work:

```
Request URL:https://api-dev.powerli.ne/api/representatives/info/0/103108
Request Method:GET
Status Code:500 Internal Server Error

{"code":500,"message":"[Semantical Error] line 0, col 63 near 'storageId = ': Error: Class Civix\\CoreBundle\\Entity\\Representative has no field or association named storageId","errors":null}
```

**Comment by igaponov** [comment]
Sorry, wasn't deployed

**Comment by petervojtek** [comment]
@igaponov , now I can access representative detail successfully.
however when I try to see representatives "Comitee membership" or "sponsored bills", I face this:
```
Request URL:https://api-dev.powerli.ne/api/representatives/info/committee/130827
Request Method:GET
Status Code:500 Internal Server Error

Request URL:https://api-dev.powerli.ne/api/representatives/info/sponsored-bills/130827
Request Method:GET
Status Code:500 Internal Server Error

{"code":500,"message":"Unable to guess how to get a Doctrine instance from the request information.","errors":null}
```

**Comment by jterps08** [comment]
For what it's worth... Committee Membership and Sponsored Bills **may** have been provided by Sunlight API in the past @igaponov

**Comment by petervojtek** [comment]
ok, it works now.. though for all the representatives I tried the Commitee membership and Sponsored bills the resultset is empty.

**Comment by jterps08** [comment]
I have to reopen this - I'm not sure why My Reps are not loading... @igaponov @devotebest please work together on this.

**Comment by igaponov** [comment]
@jterps08 check now, user user `jnq@test.com` or add a new one.

**Comment by jterps08** [comment]
Confirmed closed
